dss/stepfunctions/s3copyclient/implementation.py

```diff
@@ -111,14 +111,28 @@ def run(self) -> dict:
                 state[_Key.LAST_PART] - state[_Key.NEXT_PART] + 1))
 
             state_lock = threading.Lock()
+            complete = dict()
+            for ix in range(state[_Key.NEXT_PART], state[_Key.LAST_PART] + 1):
+                # assume things are done.
+                complete[ix] = True
+            for part_id in queue:
+                # ... except for the parts that are missing.
+                complete[part_id] = False
 
             with futures.ThreadPoolExecutor(max_workers=CONCURRENT_REQUESTS) as executor:
                 def copy_one_part(part_id):
                     self.copy_one_part(part_id)
                     with state_lock:
-                        if part_id >= state[_Key.NEXT_PART]:
+                        complete[part_id] = True
+                        for scan_part_id in range(part_id, state[_Key.LAST_PART] + 1):
```

Comment:
what is the point of this loop if you aren't using `scan_part_id`?

Reply:
Derp, that is broken.

Reply:
Oops, I missed that.